pkg/controller/controller_instance.go

```diff
@@ -208,7 +208,8 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 				return fmt.Errorf("Couldn't create a key for object %+v: %v", instance, err)
 			}
 			c.pollingQueue.Add(key)
-		} else {
+		} else if respCode == http.StatusCreated || respCode == http.StatusOK {
```

Comment:
@vaikas-google good catch. The test code is using the fake broker client instead of the real one, and the fake doesn't check for response codes like the real one does.

I'd like to keep the test intact, so I'm going to see how hard it is to retrofit a fake broker server so we can use the real client here.